--- a/src/common/restricted_token.c
+++ b/src/common/restricted_token.c
@@ -56,5 +56,7 @@
 	ZeroMemory(&si, sizeof(si));
 	si.cb = sizeof(si);
+	si.dwFlags |= STARTF_USESHOWWINDOW;
+	si.wShowWindow = SW_HIDE;
 
 	/* Open the current token to use as a base for the restricted one */
 	if (!OpenProcessToken(GetCurrentProcess(), TOKEN_ALL_ACCESS, &origToken))
@@ -105,7 +107,7 @@
 							 NULL,
 							 NULL,
 							 TRUE,
-							 CREATE_SUSPENDED,
+							 CREATE_SUSPENDED | CREATE_NO_WINDOW,
 							 NULL,
 							 NULL,
 							 &si,
--- a/src/bin/pg_ctl/pg_ctl.c
+++ b/src/bin/pg_ctl/pg_ctl.c
@@ -1779,6 +1779,8 @@
 
 	ZeroMemory(&si, sizeof(si));
 	si.cb = sizeof(si);
+	si.dwFlags |= STARTF_USESHOWWINDOW;
+	si.wShowWindow = SW_HIDE;
 
 	/*
 	 * Set stdin/stdout/stderr handles to be inherited in the child process.
@@ -1841,7 +1843,7 @@
 	}
 
 	AddUserToTokenDacl(restrictedToken);
-	r = CreateProcessAsUser(restrictedToken, NULL, cmd, NULL, NULL, TRUE, CREATE_SUSPENDED, NULL, NULL, &si, processInfo);
+	r = CreateProcessAsUser(restrictedToken, NULL, cmd, NULL, NULL, TRUE, CREATE_SUSPENDED | CREATE_NO_WINDOW, NULL, NULL, &si, processInfo);
 
 	if (IsProcessInJob(processInfo->hProcess, NULL, &inJob))
 	{
